Assets/Scripts/UI/Menu.cs

```diff
@@ -0,0 +1,28 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace OperationBlackwell.UI {
+	public class Menu : MonoBehaviour {
+
+		protected GameObject canvas_;
+		protected GameObject mainMenu_;
+		protected GameObject optionsMenu_;
+
+		private void Awake() {
+			this.canvas_ = GameObject.Find("Canvas");
+			this.mainMenu_ = canvas_.transform.Find("MainMenu").gameObject;
+			this.optionsMenu_ = canvas_.transform.Find("OptionsMenu").gameObject;
+		}
+
+		public virtual void ReturnOrExit() {
+			throw new System.NotImplementedException();
```

Comment:
Please don't use `exceptions` when not nessesary, just make a `Debug.Log` statement for this to reduce overhead in the garbagecollector.

Reply:
I don't want to log a message, I want to report a failure by the programmer. `Debug.Log` gets ignored easily, and as this throws hard errors it is more noticeable. This should be fine, and if not, I'd like to see some proof backing this up, do you have a link to a source that the GC will choke on this?

Reply:
Here is a video talking about it: https://www.youtube.com/watch?v=8BnPCav-vfc or the reference used in the video https://docs.unity3d.com/2020.2/Documentation/Manual/UnderstandingAutomaticMemoryManagement.html . Everything the GC needs to do slows the framerate, So actually both of our solutions the exception or the Debug.Log are bad for that and slows the framerate of the game. So i suggest changing it to something else that does not need to be collected.

Reply:
Hmm, interesting stuff to read. I’ll read up on it and see if I can come up with a better solution, thanks for the info.

Reply:
I've replaced the throw with a `Debug.Log`, we can change this at a later date.